src/components/Tooltip.vue

```diff
@@ -34,217 +34,343 @@ import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+                if (newContainer !== this.container) return; // container changed
+            }
+            // In case the container container is scrollable add a listener
+            await new Promise((resolve) => requestAnimationFrame(()  => {
+                if (newContainer.$el.scrollHeight !== (newContainer.$el as HTMLElement).offsetHeight) {
+                    this.container.$el.addEventListener('scroll', this.updatePosition);
+                }
+                resolve();
+            }));
         }
+
+        await this.update();
     }
 
     private mouseOver(mouseOverTooltip: boolean) {
-        if (mouseOverTooltip === false) { // mouseleave
+        if (!mouseOverTooltip) { // mouseleave
             this.mouseOverTimeout = window.setTimeout(
-                () => this._updateMouseOverTooltip(mouseOverTooltip),
+                () => this.mousedOver = false,
                 100,
             );
         } else { // mouseenter
             window.clearTimeout(this.mouseOverTimeout);
-            this._updateMouseOverTooltip(mouseOverTooltip);
+            this.mousedOver = true;
         }
     }
+}
 
-    private _updateMouseOverTooltip(mouseOverTooltip: boolean) {
-        this.update();
-        this.mousedOver = mouseOverTooltip;
+namespace Tooltip {
+    export enum Position {
+        TOP = 'top',
+        BOTTOM = 'bottom',
+    }
+
+    export enum Themes {
+        NORMAL = 'normal',
+        INVERSE = 'inverse',
     }
 }
+
+export default Tooltip;
 </script>
 
 <style scoped>
     .tooltip {
-        display: block;
+        display: inline-block;
         position: relative;
         line-height: 1;
     }
 
-    .tooltip > a {
+    .trigger {
         position: relative;
         display: block;
+        text-decoration: none;
+        outline: none;
+        cursor: default;
     }
 
-    .tooltip > a >>> svg {
+    .trigger >>> svg {
         display: block;
     }
 
-    .tooltip > a::after {
+    .trigger::after {
         opacity: 0;
         content: '';
         display: block;
         position: absolute;
-        left: calc(50% - 1rem);
-        border-width: 1rem;
-        border-style: solid;
-        transition: bottom .2s ease, top .2s ease, opacity .3s ease;
+        width: 2.25rem;
+        height: 2rem;
+        left: calc(50% - 1.125rem);
+        background: var(--nimiq-blue-bg-darkened);
+        mask-image: url('data:image/svg+xml,<svg viewBox="0 0 18 16" xmlns="http://www.w3.org/2000/svg"><path d="M9 7.12c-.47 0-.93.2-1.23.64L3.2 14.29A4 4 0 010 16h18a4 4 0 01-3.2-1.7l-4.57-6.54c-.3-.43-.76-.64-1.23-.64z" fill="white"/></svg>');
+        transition: opacity .3s ease, .3s visibility;
+        transition-delay: 16ms; /* delay one animation frame for better sync with tooltipBox */
         visibility: hidden;
         pointer-events: visible;
+        z-index: 2; /* move above tooltip-box's box-shadow */
```

Comment:
I think we can use much bigger z-index values for the tooltip, as I would expect it should always be on top. So something like `z-index: 999999` makes sense I think.